controllers/utils.go

```
@@ -53,6 +54,14 @@ func (l loggerShim) Err(msg string)     { l.logger.V(logging.InfoLevel).Info(msg
 func (l loggerShim) Crit(msg string)    { l.logger.V(logging.InfoLevel).Info(msg) }
 func (l loggerShim) Emerg(msg string)   { l.logger.V(logging.InfoLevel).Info(msg) }
 
+var caLoggerInit sync.Once
+
+func setCALogger(logger logr.Logger) {
```

Seems that we need to add such functions and package level `sync.Once` variables for each place where we want to add some fields inside functions, in order to avoid data race? IHAC that there would be too many such things. But for now, I think it is acceptable before we have got a better solution to loggings.

---

This is a special case where cfssl wants to handle its own logging instead of returning errors, so we have to stuff our own logger in. Ideally it wouldn't do that, but it's not so annoying that I'd rather write our own CA tool.

The typical case where we just call `logger := log.FromContext(ctx).WithName("MTLSCertificateCreation")` or whatever and use that doesn't need this because we're the ones handling the logs, and we use function-scoped instead of package-scoped loggers. Most libraries we use return errors that we then log, instead of using their own logging. deck is the other main exception, but that's our own damn fault.